clang/test/SemaSYCL/streams.cpp

```diff
@@ -1,107 +1,110 @@
-// RUN: %clang_cc1 -S -fsycl -fsycl-is-device -triple spir64 -ast-dump %s | FileCheck %s
+// RUN: %clang_cc1 -S -fsycl -fsycl-is-device -triple spir64 -ast-dump -sycl-std=2020 %s | FileCheck %s
 
-#include "Inputs/sycl.hpp"
+// This test demonstrates passing of SYCL stream instances as kernel arguments and checks if the compiler generates
+// the correct ast-dump
 
-template <typename Name, typename Func>
-__attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
-  kernelFunc();
-}
+#include "Inputs/sycl.hpp"
 
-using namespace cl::sycl;
+sycl::queue myQueue;
 
-handler H;
+sycl::handler H;
 
 struct HasStreams {
-  stream s1{0, 0, H};
-  stream s_array[2] = {{0, 0, H}, {0, 0, H}};
+  sycl::stream s1{0, 0, H}; // stream(totalBufferSize, workItemBufferSize, handler)
+
+  sycl::stream s_array[2] = {{0, 0, H}, {0, 0, H}};
 };
 
 struct HasArrayOfHasStreams {
   int i;
   HasStreams hs[2];
 };
 
-void use() {
-  stream in_lambda{0, 0, H};
-  stream in_lambda_array[2] = {{0, 0, H}, {0, 0, H}};
-  stream in_lambda_mdarray[2][2] = {{{0, 0, H}, {0, 0, H}}, {{0, 0, H}, {0, 0, H}}};
+int main() {
+  sycl::stream in_lambda{0, 0, H};
+  sycl::stream in_lambda_array[2] = {{0, 0, H}, {0, 0, H}};
+  sycl::stream in_lambda_mdarray[2][2] = {{{0, 0, H}, {0, 0, H}}, {{0, 0, H}, {0, 0, H}}};
 
   HasStreams Struct;
   HasArrayOfHasStreams haohs;
   HasArrayOfHasStreams haohs_array[2];
 
-  kernel<class stream_test>([=]() {
-    in_lambda.use();
-    in_lambda_array[1].use();
-    in_lambda_mdarray[1][1].use();
+  myQueue.submit([&](sycl::handler &h) {
+    h.single_task<class stream_test>([=]() {
+      in_lambda.use();
+      in_lambda_array[1].use();
+      in_lambda_mdarray[1][1].use();
 
-    Struct.s1.use();
+      Struct.s1.use();
 
-    haohs.hs[0].s1.use();
-    haohs_array[0].hs[0].s1.use();
+      haohs.hs[0].s1.use();
+      haohs_array[0].hs[0].s1.use();
+    });
   });
+
+  return 0;
 }
 
 // Function Declaration
 // CHECK: FunctionDecl {{.*}}stream_test{{.*}}
 
 // Initializers:
-// CHECK: InitListExpr {{.*}} '(lambda at
-// 'in_lambda'
-// CHECK-NEXT: CXXConstructExpr {{.*}} 'cl::sycl::stream' 'void (const cl::sycl::stream &) noexcept'
-// CHECK-NEXT: ImplicitCastExpr {{.*}} 'const cl::sycl::stream' lvalue <NoOp>
-// CHECK-NEXT: DeclRefExpr {{.*}} 'cl::sycl::stream' lvalue ParmVar
+
+// CHECK: InitListExpr {{.*}}
```

Comment:
Is there a reason you removed the `lambda at` and `in_lambda` checks

Reply:
Missed adding it back while updating the `CHECK` lines. Included it back.